proposals/0000-midstream-bindists.md

```diff
@@ -0,0 +1,199 @@
+# Midstream bindists
+
+## Abstract
+
+Bindists for the Haskell toolchain have been produced by upstream (the developers of each respective tool) for
+a long time and many tools rely on these "official" bindists (e.g. GHCup and stack).
+
+We propose here that bindists are additionally built and maintained in joint effort by the projects that
+provide installation experiences, removing the hard dependency on upstream bindists entirely.
+
+## Background
+
+Historically, installers like GHCup and stack have used upstream bindists for mainly one reason: it's easy to do
+so and doesn't require further efforts.
+
+However, using upstream bindists directly is extremely rare in the Linux world of distribution. Most distributions
+build, package, test and curate binary packages themselves, not only because they have custom formats, but for
+reasons of control, trust and quality.
+
+The relationship of GHCup and bindists has also been described in the blog post
+[GHCup is not an installer](https://hasufell.github.io/posts/2023-11-14-ghcup-is-not-an-installer.html) recently.
+
+## Problem Statement
+
+From the perspective of a GHCup developer, there are several issues with relying on upstream bindists.
+
+### Platform support
+
+GHC and other tools have in the past dropped support for certain platforms either entirely or requested
+the community to step up and do the work (e.g. on GHC CI).
```

Comment:
Let's be explicit: normally in an open-source project this is the default. Someone who cares has to step up.

GHC has paid release management from WT, so perhaps a more precise version of the complaint is "the GHC team decided not to ask their paid release manager to do this but instead requested that the community do it". So there's a disagreement about priorities.

Reply:
I disagree very strongly. Dropping an entire platform is not something trivial and needs:

* a long grace period
* a clear call for help, indicating support will be dropped unless someone stepped up

None of that happened. It was rather done ad-hoc/silently. That's not something GHCup considers good user experience. So here again seem to be very different goals/priorities.

Reply:
I was talking about the decision to not put effort towards it, not the communication issue. That I agree is problematic.